codec.go

```diff
@@ -39,19 +39,25 @@ func (e *DecodeProtocolMessageFieldError) Error() string {
 // command, etc cannot be cast, returns DecodeProtocolMessageFieldError.
 // See also godoc for json.Unmarshal, which is used for underlying decoding.
 func DecodeProtocolMessage(data []byte) (Message, error) {
-	var protomsg ProtocolMessage
-	if err := json.Unmarshal(data, &protomsg); err != nil {
+	var m struct {
```

Comment:
Can you add some comments here to describe what this struct is and why we are doing this? 
Something like: `Combine the required fields for ResponseMessage, RequestMessage, and EventMessage types so we can unmarshal these fields once.`

Reply:
LGTM thanks!

I think it makes sense to leave it here for now, but don't feel strongly either way.